src/main/kotlin/org/rust/ide/presentation/Ty.kt

```diff
@@ -61,7 +61,9 @@ private fun render(
         }"
         is TyPointer -> "*${if (ty.mutability.isMut) "mut" else "const"} ${r(ty.referenced)}"
         is TyTypeParameter -> ty.name ?: anonymous
-        is TyProjection -> "<${ty.type} as ${ty.trait.name}>::${ty.target.name}"
+        is TyProjection -> "<${ty.type} as ${ty.trait.element.name ?: return anonymous}${
+        if (includeTypeArguments) formatTraitTypeArguments(ty.trait, r, includeAssoc = false) else ""
+        }>::${ty.target.name}"
```

Comment:
Maybe split this line into some separate statements?
It's rather hard to understand what happens - too many operations per line
Also a little bit weird formatting

Reply:
There are other branches that look as awful. This function needs to be cleaned up. But we can do it in another PR.

Reply:
Okay:)